### DISCUSSION TITLE
Views and Update functions

### DISCUSSION THREAD

**Discussion by ccrotty** [apache/couchdb#3081] [Q&A] [answered]
I am using couch db in a project which stores alarm information documents. 

I'm looking for suggestions for best practices (and example links if possible) to use a view or filter to filter documents based on dynamic (and multiple) filter criteria? 

For example - assuming a document of 
{
  "_id": "...",
"_rev": "...",
"timestamp": ...
"ack": false
}

 1. What would be the best method to filter alarms based on date and the acknowledgement field?  The view doesn't have the request object. I assume that I would need to use a filter function, but the doc states it works off the changes feed?

2. If I want to set the "ack" field in all documents to true - would that be done in an update function?  Any examples on how this would be done?

Thanks,
Chris

**Comment by janl** [comment, marked as answer]
Hi Chris,

you will want to use a [View](https://docs.couchdb.org/en/stable/ddocs/views/intro.html) or a [Mango](https://docs.couchdb.org/en/latest/intro/tour.html#running-a-mango-query) [query](https://docs.couchdb.org/en/stable/api/database/find.html).

Filters are a red herring for what you are trying to do.

For updating a field in all docs: you’ll have to do that yourself: fetch each doc, update the field, save the doc.

**Reply by ccrotty** [reply]
Janl, thanks for the reply!  Is there a practical limit to the number of views to have?  Would it be better to put them each in the their own design document?

**Comment by SinanGabel** [comment]
https://docs.couchdb.org/en/stable/partitioned-dbs/index.html#partitions-by-example

This may be a good use case for you.
